techtv2ovs/utils.py

```diff
@@ -0,0 +1,314 @@
+""" techtv2ovs utils """
+import logging
+import re
+import sys
+from html import unescape
+from unicodedata import normalize
+
+import MySQLdb
+import boto3
+import bonobo
+
+from django.contrib.auth.models import User
+from django.conf import settings
+from django.utils.html import strip_tags
+
+from techtv2ovs.constants import TTV_THUMB_BUCKET, TTV_VIDEO_BUCKET, ImportStatus
+from techtv2ovs.models import TechTVCollection, TechTVVideo
+from techtv2ovs.tasks import process_videofiles
+
+from ui.models import Collection, MoiraList, Video, VideoThumbnail
+
+# pylint: disable=attribute-defined-outside-init,broad-except,too-many-instance-attributes,too-many-arguments
+
+
+def mysql_query(query, params, connection):
+    """
+    Query MySQL and return results.
+
+    Args:
+        query (str): The query string
+        params (tuple): parameters for the query string
+        connection (MySQLdb.cursors.Cursor): A MySQL cursor
+
+    Returns:
+        tuple: Query results as a tuple of tuples
+    """
+    connection.execute(query, params)
+    return connection.fetchall()
+
+
+def moiralist_name(cid, name):
+    """
+    Return a formatted moira list name based on the name of a collection
+    Based on TechTV / Ruby 'parameterize' function
+    https://github.com/TechTV/TechTV/blob/master/app/models/collection/moira_controllable.rb#L76
+
+    Args:
+        cid (int): The ID of a TechTVCollection object
+        name (str): The name of the TechTVCollection object
+
+    Returns:
+        str: The name of the moira list for the collection
+    """
+    # Replace certain special characters with dashes
+    return 'techtv-{}-{}'.format(
+        cid,
+        re.sub(r'\-+', '-', (re.sub(r'[^A-Za-z0-9\-_]+', '', re.sub(r'[\s\'\.,\-\/\\@]+', '-', name))))[0:32].lower()
+    )
+
+
+def get_s3_videos(s3_prefix):
+    """
+    Get all the available S3 videofile objects for a TechTVVideo
+
+    Args:
+        s3_prefix (str): The TechTVVideo external id to use as the S3 bucket prefix to filter by
+
+    Returns:
+        list: A list of S3 objects
+    """
+    s3client = boto3.client('s3')
```

Comment:
It looks like this function is uncovered, can you write a test for it?

Reply:
✅